### PR TITLE
Modified to allow selection of Center Project items to include the fo…

### PR DESCRIPTION
**Pull request by Ohsudev**
…llowing

entries:
center project, investigator name, and alias number.  This info should be displayed only for monkey ids assigned to a project.

#### Changes
Modified to allow selection of Center Project items to include the following
entries:
center project, investigator name, and alias number.  This info should be displayed only for monkey ids assigned to a project.

### CODE REVIEW

File: onprc_ehr/resources/web/onprc_ehr/form/field/ProjectEntryField.js [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+            sql += "SELECT p.project as project, p.displayName as displayName, p.account as account, p.protocol.displayName as protocolDisplayName, " +
+                    " p.protocol as protocol, p.title, p.shortname, p.investigatorId." + this.invesLastNameCol + " as investigator, " +
+                    " CASE WHEN (a.project = p.project AND p.use_category = 'Research') THEN 0 WHEN (a.project = p.project) THEN 1 ELSE 2 END as sort_order, " +
+                    " CASE WHEN (a.project = p.project) THEN 1 ELSE 0 END as isAssigned " +
+                    " FROM ehr.project p JOIN study.assignment a ON (a.project.protocol = p.protocol) " +
+                    " WHERE a.id='"+id+"' AND (a.project = p.project) "; //TODO: restore this OR p.enddate IS NULL OR p.enddate >= curdate()
+
+            //NOTE: if the date is in the future, we assume active projects
+            if (date){
+                sql += "AND cast(a.date as date) <= '"+date.format('Y-m-d')+"' AND ((a.enddateCoalesced >= '"+date.format('Y-m-d')+"') OR ('"+date.format('Y-m-d')+"' >= now() and a.enddate IS NULL))";
```
Comment (labkey-jeckels) [inline onprc_ehr/resources/web/onprc_ehr/form/field/ProjectEntryField.js]:
```suggestion
        sql += "AND cast(a.date as date) <= '"+Ext4.Date.format(date, 'Y-m-d')+"' AND ((a.enddateCoalesced >= '"+Ext4.Date.format(date, 'Y-m-d')+"') OR ('"+Ext4.Date.format(date, 'Y-m-d')+"' >= now() and a.enddate IS NULL))";
```

File: onprc_ehr/resources/web/onprc_ehr/model/sources/BiopsyNotes.js [hunk 2]
```diff
@@ -16,10 +16,14 @@ EHR.model.DataModelManager.registerMetadata('Biopsy_Notes', {
...
             project: {
-                xtype: 'ehr-projectentryfield'
+                xtype: 'onprc_ehr-projectentryfield'
             }
         },
 
-
+        'study.pathologyDiagnoses': {
+            codesRaw: {
+                header: 'Snomed Codes'
```
Comment (labkey-jeckels) [inline onprc_ehr/resources/web/onprc_ehr/model/sources/BiopsyNotes.js:25]: Is this intentional? If so, no problem, it's just that it's unrelated to the rest of the changes.

**Review by labkey-jeckels — CHANGES_REQUESTED** [review]
One suggested edit to avoid ExtJS compatibility problems.

File: onprc_ehr/resources/web/onprc_ehr/form/field/ProjectEntryField.js (hunk 1 above)
Comment (Ohsudev) [reply onprc_ehr/resources/web/onprc_ehr/form/field/ProjectEntryField.js]: The previous code allowed the Investigator name to be displayed, whereas the new coding format that your provided prevents the investigator name from being displayed.
This is the only reason I created a new onprc_ehr version of the same file name.

Is there an alternative code that you can suggest that allows the investigator name to be displayed?

Raymond

File: onprc_ehr/resources/web/onprc_ehr/model/sources/BiopsyNotes.js (hunk 2 above)
Comment (Ohsudev) [reply onprc_ehr/resources/web/onprc_ehr/model/sources/BiopsyNotes.js:25]: Thanks, Josh.  Yes, the header as it appears in this code is what the customer requested to be displayed.

File: onprc_ehr/resources/web/onprc_ehr/form/field/ProjectEntryField.js (hunk 1 above)
Comment (Ohsudev) [reply onprc_ehr/resources/web/onprc_ehr/form/field/ProjectEntryField.js]: Sorry, Josh. i left out some code.  The code you provided works very well as designed. I will resubmit the changes to the program again to you

**Comment by Ohsudev** [comment]
I have made the codes changes you requested.  Please review code again for me.

**Review by labkey-jeckels — APPROVED** [review]
Looks good, thanks.

**Pull request merged**